Feature/api get lexicon name

AlejandroRG31 opened the pull request:
Added a function to allow get all the lexicon names using the api methods 
Added a serializer of lexicon which has as values/variables: name source-language and dest-language
Added a test to check that this api method works correctly

AlejandroRG31 committed:
Added serializer lexicon and api method to get all the lexicon names

AlejandroRG31 committed:
Added lexicon api test

slamora commented:
While I was reviewing your code, I wonder if we could refactor `Lexicon` model to have a human code to identify it in a unique way and besides that keep a verbose name. Something like:
```python
class Lexicon(models.Model):
    code = models.SlugField(primary=True)
    name = models.CharField(max_length=32)
    description = models.TextField(blank=True)
    # Define languages using ISO 639-1 values
    src_language = models.CharField(max_length=2)
    dst_language = models.CharField(max_length=2)
```
**NOTE**: untested code .
**NOTE2** `code` slug may be generated automatically combining `src_language`and `dst_language` --> `{src}-{dst}`. E.g. `es-ar`

@AlejandroRG31 What do you think?

slamora commented on linguatec_lexicon/views.py:
```diff
@@ -80,6 +80,25 @@ class DefaultLimitOffsetPagination(LimitOffsetPagination):
+    serializer_class = LexiconSerializer
+    pagination_class = DefaultLimitOffsetPagination
+
+    @action(detail=False)
+    def get_lexicon_names(self, request):
+        queryset = Lexicon.objects.all().order_by('name')
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
+
```
I think that this method duplicates the response provided by default `list()` method.
If you need to customize queryset order_by you could override `get_queryset()` method.

AlejandroRG31 replied:
Yeah this method was like an overwritting of the default method, i have deleted this and refactored the default method

slamora commented on tests/test_api.py:
```diff
@@ -29,6 +29,16 @@ def test_word_search_no_results(self):
+    fixtures = ['lexicon-sample.json']
+
+    def test_lexicon(self):
+        resp = self.client.get('/api/lexicon/get_lexicon_names/')
```
Standar API REST will expects that URL was `/api/lexicon/` OR better in plural `/api/lexicons/`.

AlejandroRG31 replied:
API refactored to use the default method and avoid using the overwritten one, also changed the default method name from lexicon to lexicons as you requested, which i also think is better and more intuitive

AlejandroRG31 committed:
Modified get lexicons method in api and created a calculated variable code in lexicon.

Modified get lexicons method because the default method was already doing the work
of returning the lexicons as requested.

Variable or property code added to lexicon model which is a string of
src_language + '-' + dst_language

AlejandroRG31 committed:
Revert "Modified get lexicons method in api and created a calculated variable code in lexicon."

This reverts commit 650355c167142bbd4cbfbf4e52a3b1e942023e28.

AlejandroRG31 committed:
Deleted in API an unnecesary method that was overwritting the default method when both were doing the same

AlejandroRG31 commented:
> While I was reviewing your code, I wonder if we could refactor `Lexicon` model to have a human code to identify it in a unique way and besides that keep a verbose name. Something like:
> 
> ```python
> class Lexicon(models.Model):
>     code = models.SlugField(primary=True)
>     name = models.CharField(max_length=32)
>     description = models.TextField(blank=True)
>     # Define languages using ISO 639-1 values
>     src_language = models.CharField(max_length=2)
>     dst_language = models.CharField(max_length=2)
> ```
> 
> **NOTE**: untested code .
> **NOTE2** `code` slug may be generated automatically combining `src_language`and `dst_language` --> `{src}-{dst}`. E.g. `es-ar`
> 
> @AlejandroRG31 What do you think?

I think that a solution could be:

-Create a property or calculated value
-Make src_language  and dst_language unique together variables in order to make code also an unique variable
-Don't change name, in case we want to use name to save the name of the version, like "name = es-ar (version 1.1)"

```
@property
    def code(self):
        return (self.src_language + '-' + self.dst_language)

    name = models.CharField(unique=True, max_length=32)
   
    src_language = models.CharField(max_length=2)
    dst_language = models.CharField(max_length=2)

    unique_together = [['src_language', 'dst_language']]
```

The only problem is that code can not be used as a varible to filter, get .... (lexicon.object.get(code='ar-es')) but it can be used as a secondary name.

AlejandroRG31 committed:
Refactored Word.search_near and API near to use lexicon name as search method already do

AlejandroRG31 committed:
Merge branch 'master' into feature/api-get-lexicon-name

slamora commented:
@AlejandroRG31 we need to be able to filter by Lexicon. So instead of implementing it as a `@property`  let's store it on the database.

Anyway, I agree with your proposal of `unique_together(src_language, dst_language)`.

slamora commented:
@AlejandroRG31 I've seen that proposed changes are implemented on #10 . So, I guess that this PR could be merged and closed.
